ietf/doc/templatetags/ietf_filters.py

```diff
@@ -613,3 +613,89 @@ def action_holder_badge(action_holder):
     else:
         return ''  # no alert needed
 
+@register.filter
+def is_regular_agenda_item(assignment):
+    """Is this agenda item a regular session item?
+    
+    A regular item appears as a sub-entry in a timeslot within the agenda
+
+    >>> from collections import namedtuple  # use to build mock objects
+    >>> mock_timeslot = namedtuple('t2', ['slug'])
+    >>> mock_assignment = namedtuple('t1', ['slot_type'])  # slot_type must be a callable
+    >>> factory = lambda t: mock_assignment(slot_type=lambda: mock_timeslot(slug=t))
+    >>> is_regular_agenda_item(factory('regular'))
+    True
+
+    >>> any(is_regular_agenda_item(factory(t)) for t in ['plenary', 'break', 'reg', 'other', 'officehours'])
+    False
+    """
+    return assignment.slot_type().slug == 'regular'
+
+@register.filter
+def is_plenary_agenda_item(assignment):
+    """Is this agenda item a regular session item?
+    
+    A regular item appears as a sub-entry in a timeslot within the agenda
+
+    >>> from collections import namedtuple  # use to build mock objects
+    >>> mock_timeslot = namedtuple('t2', ['slug'])
+    >>> mock_assignment = namedtuple('t1', ['slot_type'])  # slot_type must be a callable
+    >>> factory = lambda t: mock_assignment(slot_type=lambda: mock_timeslot(slug=t))
+    >>> is_plenary_agenda_item(factory('plenary'))
+    True
+
+    >>> any(is_plenary_agenda_item(factory(t)) for t in ['regular', 'break', 'reg', 'other', 'officehours'])
+    False
+    """
+    return assignment.slot_type().slug == 'plenary'
+
+@register.filter
+def is_special_agenda_item(assignment):
+    """Is this agenda item a special item?
+    
+    Special items appear as top-level agenda entries with their own timeslot information.
+
+    >>> from collections import namedtuple  # use to build mock objects
+    >>> mock_timeslot = namedtuple('t2', ['slug'])
+    >>> mock_assignment = namedtuple('t1', ['slot_type'])  # slot_type must be a callable
+    >>> factory = lambda t: mock_assignment(slot_type=lambda: mock_timeslot(slug=t))
+    >>> all(is_special_agenda_item(factory(t)) for t in ['break', 'reg', 'other', 'officehours'])
+    True
+
+    >>> any(is_special_agenda_item(factory(t)) for t in ['regular', 'plenary'])
+    False
+    """
+    return assignment.slot_type().slug in [
+        'break',
+        'reg',
+        'other',
+        'officehours',
+    ]
+
+@register.filter
+def should_show_agenda_session_buttons(assignment):
+    """Should this agenda item show the session buttons (jabber link, etc)?
```

Comment:
I've renamed it to `should_suppress_agenda_session_buttons`. I would like to avoid tying its name to its current logic. I've added the comments to clarify that it's for use in the `session_buttons_include.html` template.

Also, I'm moving this and the various `is_*_agenda_item` out of the `ietf/doc/templatetags/ietf_filters.py` file and into `ietf/meeting/templatetags/agenda_custom.py`. These methods have nothing to do with docs (I had assumed `ietf_filters.py` was a utils module, hadn't noticed it was in doc until recently).

Reply:
Oh, I hadn't even notice that directory discrepancy!  Good catch.